src/lib.rs

```diff
@@ -4,52 +4,57 @@
 extern crate delog;
 generate_macros!();
 
-pub use heapless_bytes::Bytes;
-
 pub(crate) mod consts;
 pub mod de;
 pub mod error;
 pub mod ser;
 
 pub use error::{Error, Result};
+use ser::Writer;
 
-// pub use de::from_bytes;
-// pub use de::take_from_bytes;
+/// Serialize an object to a `Writer`
+///
+/// Returns the amount of bytes written to the writer
+pub fn cbor_serialize_to<'a, T: ?Sized + serde::Serialize, W: Writer>(
+    object: &T,
+    writer: W,
+) -> Result<usize> {
+    let mut serializer = ser::Serializer::new(writer);
+    object.serialize(&mut serializer)?;
+    Ok(serializer.written())
+}
 
 // kudos to postcard, this is much nicer than returning size
 pub fn cbor_serialize<'a, T: ?Sized + serde::Serialize>(
     object: &T,
     buffer: &'a mut [u8],
 ) -> Result<&'a [u8]> {
-    let writer = ser::SliceWriter::new(buffer);
-    let mut ser = ser::Serializer::new(writer);
-
-    object.serialize(&mut ser)?;
-
-    let writer = ser.into_inner();
-    let size = writer.bytes_written();
-
-    Ok(&buffer[..size])
+    let mut buf = &mut *buffer;
+    let written = cbor_serialize_to(object, &mut buf)?;
+    Ok(&buffer[..written])
 }
 
+#[cfg(feature = "heapless-bytes-v0-3")]
 /// Append serialization of object to existing bytes, returning length of serialized object.
 pub fn cbor_serialize_extending_bytes<T: ?Sized + serde::Serialize, const N: usize>(
```

Comment:
Should we deprecate this function in favor of `cbor_serialize_to`?  It would mean that we could need additional type annotations in some cases, but it is somewhat inconsistent to only provide this function for heapless-bytes 0.3.

We don’t need to do it in this PR but should at least create an issue for it if we don’t.

Reply:
I was just thinking of adding the `#[deprecated]` attribute.  We can then remove it later.

Reply:
I think I would:

- merge this as is (so we don't get warnings in our current releases), and remove it for the 0.5.0 release.

Reply:
If you want a compatible intermediate state, we should also add `heapless-bytes-v0-3` to the default features (and remove it for the release).